tshirts.cpp

```diff
@@ -1,22 +1,52 @@
 #include <assert.h>
 #include <iostream>
 
-char size(int cms) {
-    char sizeName = '\0';
-    if (cms < 38) {
-        sizeName = 'S';
-    } else if (cms > 38 && cms < 42) {
-        sizeName = 'M';
-    } else if (cms > 42) {
-        sizeName = 'L';
+enum class Size {
+    XS,
+    S,
+    M,
+    L,
+    XL,
+    Invalid
+};
+
+Size size(int cms) {
+    Size sizeName = Size::Invalid;
+
+    switch (cms) {
+        case 36:
+            sizeName = Size::XS;
+            break;
+        case 38:
+            sizeName =  Size::S;
+            break;
+        case 40:
+            sizeName = Size::M;
+            break;
+        case 42:
+            sizeName = Size::L;
+            break;
+        case 44:
+            sizeName = Size::XL;
+            break;
     }
     return sizeName;
 }
 
-int main() {
-    assert(size(37) == 'S');
-    assert(size(40) == 'M');
-    assert(size(43) == 'L');
+void testSize() {
+    assert(size(36) == Size::XS);
+    assert(size(38) == Size::S);
+    assert(size(40) == Size::M);
+    assert(size(42) == Size::L);
+    assert(size(44) == Size::XL);
+    assert(size(37) == Size::Invalid);

```

Comment:
People who have shoulder size with odd cms cannot buy shirt :)
Perhaps this code is for a UI where it has a drop-down with only even numbers as shirt sizes... in which case it is correct